lib/mapObjects/CRewardableObject.h

```diff
@@ -29,26 +29,26 @@ class DLL_LINKAGE CRewardLimiter
 {
 public:
 	/// day of week, unused if 0, 1-7 will test for current day of week
-	si32 dayOfWeek;
-	si32 daysPassed;
+	si32 dayOfWeek = 0;
+	si32 daysPassed = 0;
 
 	/// total experience that hero needs to have
-	si32 heroExperience;
+	si32 heroExperience = 0;
 
 	/// level that hero needs to have
-	si32 heroLevel;
+	si32 heroLevel = 0;
 
 	/// mana points that hero needs to have
-	si32 manaPoints;
+	si32 manaPoints = 0;
 
 	/// percentage of mana points that hero needs to have
-	si32 manaPercentage;
+	si32 manaPercentage = 0;
 
 	/// resources player needs to have in order to trigger reward
 	TResources resources;
 
 	/// skills hero needs to have
-	std::vector<si32> primary;
+	std::vector<si32> primary {0, 0, 0, 0};
```

Comment:
This should have GameConstants::PRIMARY_SKILLS instead of hardcoded 4 elements.
Perhaps switch to std array?

Reply:
There I need to rewrite a json load function too, and json load function can return less than 4 elements. So, what do you think should be better?

Reply:
But it can not? It will always return 4 elements?
```

	std::vector<si32> loadPrimary(const JsonNode & value, CRandomGenerator & rng)
	{
		std::vector<si32> ret;
		for(const auto & name : PrimarySkill::names)
		{
			ret.push_back(loadValue(value[name], rng));
		}
		return ret;
	}
```

Reply:
Okay, switching to std::array makes sense there. What do you think about default initializations not at all, but especially in this PR for secondary classes? I did this change only because there was one constructor in header, and I do not see why we should keep it (and when we will add another variable, we should manually keep constructor init + variable in sync).

Reply:
Keep as before. There is 0 need to change it IMO.

Spending time on. one of... 225 open issue would be way more productive than these "improvements"